rio_stac/stac.py

```diff
@@ -41,19 +42,19 @@ def get_metadata(
     """Get Raster Metadata."""
     metadata: Dict[str, Any] = {}
 
-    # To Do: handle non-geo data
     if src_dst.crs is not None:
-        bbox = warp.transform_bounds(
-            src_dst.crs, "epsg:4326", *src_dst.bounds, densify_pts=21
-        )
+        src_geom = bbox_to_geom(src_dst.bounds)
+        geom = warp.transform_geom(src_dst.crs, "epsg:4326", src_geom)
+        bbox = feature_bounds(geom)
```

Comment:
1. construct geometry using dataset bounds in input projection

2. retroject (and split if needed) the geometry in WGS84

3. get the bbox using rasterio function

Reply:
I recommend using pyproj for this:
https://pyproj4.github.io/pyproj/stable/api/transformer.html#pyproj.transformer.Transformer.transform_bounds

Or rasterio 1.3+ and GDAL 3.4+.

Related:
- https://github.com/rasterio/rasterio/issues/2222
- https://github.com/rasterio/rasterio/issues/2144
- https://github.com/rasterio/rasterio/issues/2313

Reply:
thanks @snowman2.
I'd like to avoid to add a dependency (even if I like pyproj). My understanding is that `transform_geom` should handle antimeridian splitting just fine (confirmed by the test) and the file provided by @robintw was kinda a corner case.

Reply:
That makes sense. One drawback of that approach is you miss out on the benefits of using `transform_bounds`: "Transform boundary densifying the edges to account for nonlinear transformations along these edges and extracting the outermost bounds."

To get around that, you could potentially densify the points in the geometry before passing it to GDAL.